R/utils-read.R

```diff
@@ -9,6 +9,31 @@ sizeOf <- function(df){
   return(nrow(df))
 }
 
+#' Returns whether or not all required columns are present
+#' 
+#' 
+emptyData <- function(data, name, requiredFields){
+  if(!isEmptyOrBlank(data)){
+    if(!all(requiredFields %in% names(data)) || any(is.na(data[requiredFields]))){
+      #Checking returned JSON structure
+      missingFields <- requiredFields[!requiredFields %in% names(data)]
+      
+      #Chceking JSON array entries for consistency of required fields
+      partialFields <- ifelse(is.data.frame(data), names(which(colSums(is.na(data)) > 0)), "")
+      partialFields <- partialFields[which(names(partialFields) %in% requiredFields)]
+      
+      
+      naCols <- c(missingFields, partialFields)
+      stop(paste("Report JSON for requested field: ", name, " is missing required fields: {",  paste(naCols, collapse=', '), "}"))
+    } else {
+      return(FALSE)
+    }
+  } else if(is.null(data)){
+    stop(paste("Requested field: '", name, "' not found in report JSON."))
```

Comment:
Rather than pass the name function in for the purpose of labeling the errors, we should leave it up to the caller to make the error more specific.

I think we have the same pattern going on in validParam, I will update that as well.